src/components/App.js

```diff
@@ -1,17 +1,166 @@
-import React from 'react';
-
-class App extends React.Component {
-  constructor(){
-    super();
-  }
-
-  render(){
-    return (
-      <div>
-        React cinema app
-      </div>
-    )
-  }
-}
-
-export default App;
+//API key: 210776d9
+
+//http://www.omdbapi.com/?apikey=210776d9&[searchparameters]
+import React from 'react';
+import HeaderSearch from "./HeaderSearch.js"
+import SearchDisplay from "./SearchDisplay.js"
+import FilmDisplay from "./FilmDisplay.js"
+import HomePage from "./HomePage.js"
+
+class App extends React.Component {
+  constructor(){
+    super();
+    this.state = {
+      searchString: "",
+      searchResults: [],
+      searchPage: 1,
+
+      currentFilmId: "",
+      currentFilm: {},
+
+      favourites: [],
+      favouriteResults: [],
+
+      display: 'home' // home, film, search
+
+    }
+
+
+    this.fetchSearchResults = this.fetchSearchResults.bind(this)
+    this.retrieveSearchString = this.retrieveSearchString.bind(this)
+    this.retrieveFilmId = this.retrieveFilmId.bind(this)
+    this.toggleDisplay = this.toggleDisplay.bind(this)
+    this.retrieveFave = this.retrieveFave.bind(this)
+    this.fetchFavourite = this.fetchFavourite.bind(this)
+    this.populateFavourites = this.populateFavourites.bind(this)
+
+  }
+
+  componentDidMount(){
+    this.populateFavourites()
+
+
+
+  }
+
+
+
+  populateFavourites(){
+    const faveList = JSON.parse(localStorage.getItem("favourites"))
+    this.setState({
+      favourites: faveList
+    })
+    //pretty sure this isnt the best way of doing this, but my favouritesResults was getting duplicated
+    //when Homepage mounted
+    this.setState({
+      favouriteResults: []
+    })
+    //make sure we received an array in local storage before using array method to fill favouriteResults
+    faveList

```

Comment:
Since we are not using the last part of the ternary, an `if` statement might be better here